<launch>
  <!-- Global parameters -->
  <rosparam>
    thesis/rgb_image_topic:   'kinect_head/rgb/image_rect_color'
    thesis/depth_image_topic: 'kinect_head/depth_registered/image_rect'
    thesis/camera_info_topic: 'kinect_head/depth_registered/camera_info'
    thesis/camera_frame:      '/head_mount_kinect_rgb_optical_frame' <!-- /head_mount_kinect_rgb_optical_frame -->
    thesis/map_frame:         '/map'
    thesis/tf_timeout:        2.5
    thesis/openni_timeout:    5.0
    thesis/memory_size:       64
  </rosparam>

  <!-- Database -->
  <node pkg="thesis" type="init_database" name="thesis_database" output="screen">
    <param name="debug"      type="bool" value="true"/>
    <param name="image_path" type="str"  value="$(find thesis)/img/"/>
  </node>
  
  <!-- Perception -->
  <node pkg="thesis" type="init_recognition" name="thesis_recognition" output="screen">
    <param name="debug"                 type="bool"   value="true"/>
    <param name="mipmap_level"          type="int"    value="1"/>
    <param name="max_objects_per_frame" type="int"    value="5"/>
    <param name="max_nof_keypoints"     type="int"    value="256"/>
    <param name="knn_1to2_ratio"        type="double" value="0.9"/>
    <param name="fov_width"             type="int"    value="60"/>
    <param name="fov_height"            type="int"    value="45"/>
    <!-- Kinect range limits -->
    <param name="fov_near"              type="double" value="1.2"/>
    <param name="fov_far"               type="double" value="3.5"/>
  </node>
  
  <!-- Semantic map -->
  <node pkg="thesis" type="init_mapping" name="thesis_mapping" output="screen">
    <param name="debug"             type="bool"   value="true"/>
    <param name="age_threshold"     type="double" value="2.0"/>
    <param name="min_confirmations" type="int"    value="3"/>
  </node>
  
  <!-- Visualisation -->
  <node pkg="thesis" type="init_visualization" name="thesis_visualization" output="screen">
    <param name="debug" type="bool" value="false"/>
  </node>
</launch>
